### PR TITLE
Clean up the description of tree evolution.

### PR DESCRIPTION
**Pull request by MartaMularczyk**
Changes (apart from editorial):
* Remove the text about the key schedule. Reason: I found the text confusing, because updating paths provides FS and PCS for the HPKE keys in the ratchet tree, and not directly for the key schedule.
* Generate the leaf key pair simply as a fresh HPKE key pair, and not from the leaf_secret. I couldn't see any benefit of using the leaf_secret. I wonder if there is a good reason?

### CODE REVIEW

File: draft-ietf-mls-protocol.md [hunk 1]
```diff
@@ -1504,51 +1504,42 @@ member of the group.
...
+all HPKE key pairs in the ratchet tree for which they know the secret keys.
+More precisely, they refresh the key pairs of their leaf and of all nodes on
+their leaf's direct path.
 
-To begin, the generator of the UpdatePath updates its leaf
-KeyPackage and its direct path to the root with new secret values.  The
-HPKE leaf public key within the KeyPackage MUST be derived from a freshly
-generated HPKE secret key to provide post-compromise security.
+The member initiating the epoch change generates the fresh key pairs using the
+following procedure. The procedure is designed in a way that allows to
```
Comment (bifurcation) [inline draft-ietf-mls-protocol.md:1522]: "allows" needs a subject here.

```suggestion
following procedure. The procedure is designed in a way that allows group members to
```

File: draft-ietf-mls-protocol.md [hunk 2]
```diff
@@ -1504,51 +1504,42 @@ member of the group.
...
-"leaf_secret", and uses the leaf_secret to generate their leaf HPKE key pair
-(see {{key-packages}}) and to seed a sequence of "path secrets", one for each
-ancestor of its leaf. In this setting,
-path_secret\[0\] refers to the leaf's parent,
-path_secret\[1\] to the parent's parent, and so on. At each step, the path
-secret is used to derive a new secret value for the corresponding
-node, from which the node's key pair is derived.
+* Generate a fresh HPKE key pair for the leaf.
+* Generate a sequence of path secrets, one for each node on the leaf's direct
+path as follows. Sample a fresh random value `path_secret[0]` and compute
```
Comment (bifurcation) [inline draft-ietf-mls-protocol.md:1528]: I would split the middle bullet here:

* Generate the leaf key pair
* Generate path secrets
    * `path_secret[0]` is sampled at random
    * `path_secret[n] = DeriveSecret(path_secret[n-1], "path")`
* Derive node key pairs from path secrets

**Review by bifurcation — APPROVED** [review]
A couple of editorial comments, but in general, LGTM

File: draft-ietf-mls-protocol.md (hunk 2 above)
Comment (bifurcation) [reply draft-ietf-mls-protocol.md:1528]: Note also that the current version doesn't render quite properly.  "In this context..." isn't rendered as part of the bullet.  You might be able to fix this by indenting that part by 2 spaces.  Or you might have better success putting the code block after the prose.

**Comment by rohan-wire** [comment]
> * Generate the leaf key pair simply as a fresh HPKE key pair, and not from the leaf_secret. I couldn't see any benefit of using the leaf_secret. I wonder if there is a good reason?

I am concerned that we are asking for a LOT of fresh entropy if (in addition to nonces) we need a crypto-random value every epoch for every group (a Wire client can easily a member of thousands of groups at a time). Do you have a concern about deriving it?

**Comment by bifurcation** [comment]
Discussion on the virtual interim:

* To @rohan-wire's concern: This allows the base path secret to be derived, e.g., using a KDF, so this doesn't *necessarily* increase the amount of entropy needed
* Clear to merge

**Review by bifurcation — APPROVED** [review]

**Pull request merged**